### ISSUE TITLE
androidNativeActivity sample unable to build

### ISSUE THREAD

**Issue by altamic** [JetBrains/kotlin-native#831]
I get this error from the linker:

`arm-linux-androideabi/bin/ld: error: undefined symbol 'Konan_main' referenced in expression`

Here the details:
```
altamic@phobos:kotlin_native_dir $ pwd
/Users/altamic/projects/work/kotlin_native_dir
altamic@phobos:kotlin_native_dir $ uname -a
Darwin phobos 15.6.0 Darwin Kernel Version 15.6.0: Tue Apr 11 16:00:51 PDT 2017; root:xnu-3248.60.11.5.3~1/RELEASE_X86_64 x86_64
altamic@phobos:kotlin_native_dir $ ls
kotlin-native-macos-0.3.1
altamic@phobos:kotlin_native_dir $ echo $ANDROID_HOME
/usr/local/Cellar/android-sdk/24.3.4
altamic@phobos:kotlin_native_dir $ android list target | grep -A2 25
id: 7 or "android-25"
     Name: Android API 25
     Type: Platform
     API level: 25
     Revision: 3
----------
altamic@phobos:kotlin_native_dir $ cd kotlin-native-macos-0.3.1/
altamic@phobos:kotlin-native-macos-0.3.1 $ cd samples/androidNativeActivity/
altamic@phobos:androidNativeActivity $ ../gradlew buildApk
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
Parallel execution is an incubating feature.
:downloadKonanCompiler
:genArm32InteropStubs
:compileArm32InteropStubs
KtFile: android.kt
manifest addend: // headers
:compileKonanPolyhedronArm32
KtFile: engine.kt
KtFile: polyhedra.kt
KtFile: renderer.kt
KtFile: vectors.kt
/Users/altamic/projects/work/kotlin_native_dir/kotlin-native-macos-0.3.1/samples/androidNativeActivity/../../dependencies/target-toolchain-21-osx-android_arm32/bin/../lib/gcc/arm-linux-androideabi/4.9.x/../../../../arm-linux-androideabi/bin/ld: 
error: undefined symbol 'Konan_main' referenced in expression
clang38: error: linker command failed with exit code 1 (use -v to see invocation)
exception: java.lang.IllegalStateException: The /Users/altamic/projects/work/kotlin_native_dir/kotlin-native-macos-0.3.1/samples/androidNativeActivity/../../dependencies/target-toolchain-21-osx-android_arm32/bin//clang command returned non-zero exit code: 1.
        at org.jetbrains.kotlin.backend.konan.LinkStage.runTool(LinkStage.kt:352)
        at org.jetbrains.kotlin.backend.konan.LinkStage.link(LinkStage.kt:305)
        at org.jetbrains.kotlin.backend.konan.LinkStage.access$link(LinkStage.kt:186)
        at org.jetbrains.kotlin.backend.konan.LinkStage$linkStage$3.invoke(LinkStage.kt:376)
        at org.jetbrains.kotlin.backend.konan.LinkStage$linkStage$3.invoke(LinkStage.kt:186)
        at org.jetbrains.kotlin.backend.konan.PhaseManager$phase$$inlined$with$lambda$1.invoke(KonanPhases.kt:130)
        at org.jetbrains.kotlin.backend.konan.PhaseManager$phase$$inlined$with$lambda$1.invoke(KonanPhases.kt:109)
        at org.jetbrains.kotlin.backend.konan.util.UtilKt.profileIf(util.kt:33)
        at org.jetbrains.kotlin.backend.konan.PhaseManager.phase$backend_native_compiler(KonanPhases.kt:129)
        at org.jetbrains.kotlin.backend.konan.LinkStage.linkStage(LinkStage.kt:375)
        at org.jetbrains.kotlin.backend.konan.KonanDriverKt$runTopLevelPhases$5.invoke(KonanDriver.kt:111)
        at org.jetbrains.kotlin.backend.konan.KonanDriverKt$runTopLevelPhases$5.invoke(KonanDriver.kt)
        at org.jetbrains.kotlin.backend.konan.PhaseManager$phase$$inlined$with$lambda$1.invoke(KonanPhases.kt:130)
        at org.jetbrains.kotlin.backend.konan.PhaseManager$phase$$inlined$with$lambda$1.invoke(KonanPhases.kt:109)
        at org.jetbrains.kotlin.backend.konan.util.UtilKt.profileIf(util.kt:33)
        at org.jetbrains.kotlin.backend.konan.PhaseManager.phase$backend_native_compiler(KonanPhases.kt:129)
        at org.jetbrains.kotlin.backend.konan.KonanDriverKt.runTopLevelPhases(KonanDriver.kt:110)
        at org.jetbrains.kotlin.cli.bc.K2Native.doExecute(K2Native.kt:48)
        at org.jetbrains.kotlin.cli.bc.K2Native.doExecute(K2Native.kt:38)
        at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.java:103)
        at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.java:51)
        at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:92)
        at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:70)
        at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:36)
        at org.jetbrains.kotlin.cli.common.CLITool$Companion.doMainNoExit(CLITool.kt:157)
        at org.jetbrains.kotlin.cli.common.CLITool$Companion.doMain(CLITool.kt:148)
        at org.jetbrains.kotlin.cli.bc.K2Native$Companion$main$1.invoke(K2Native.kt:145)
        at org.jetbrains.kotlin.cli.bc.K2Native$Companion$main$1.invoke(K2Native.kt:142)
        at org.jetbrains.kotlin.backend.konan.util.UtilKt.profileIf(util.kt:33)
        at org.jetbrains.kotlin.backend.konan.util.UtilKt.profile(util.kt:28)
        at org.jetbrains.kotlin.cli.bc.K2Native$Companion.main(K2Native.kt:144)
        at org.jetbrains.kotlin.cli.bc.K2NativeKt.main(K2Native.kt:150)

:compileKonanPolyhedronArm32 FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileKonanPolyhedronArm32'.
> Process 'command '/Library/Java/JavaVirtualMachines/jdk1.8.0_60.jdk/Contents/Home/bin/java'' finished with non-zero exit value 2

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 1 mins 35.144 secs
```

it seems that `Konan_main` is referenced here: 

```
altamic@phobos:androidNativeActivity $ ag -C3 Konan_main
src/main/AndroidManifest.xml
22-      <meta-data android:name="android.app.lib_name"
23-                 android:value="poly" />
24-      <meta-data android:name="android.app.func_name"
25:                 android:value="Konan_main" />
26-      <intent-filter>
27-        <action android:name="android.intent.action.MAIN" />
28-        <category android:name="android.intent.category.LAUNCHER" />
```
and it has been introduced as `entrySelector` for different targets in `konan.properties`
```
altamic@phobos:androidNativeActivity $ ag -C3 entrySelector ../../konan/konan.properties
44-osVersionMinFlagLd.osx = -macosx_version_min
45-osVersionMinFlagClang.osx = -mmacosx-version-min
46-osVersionMin.osx = 10.11
47:entrySelector.osx = -alias _Konan_main _main
48-dependencies.osx = target-sysroot-1-darwin-macos \
49-    libffi-3.2.1-2-darwin-macos \
50-    clang-llvm-3.9.0-darwin-macos
--
57-  target-sysroot-2-darwin-ios
58-
59-arch.ios = arm64
60:entrySelector.ios = -alias _Konan_main _main
61-targetSysRoot.ios = target-sysroot-2-darwin-ios
62-libffiDir.ios = libffi-3.2.1-2-darwin-ios
63-llvmLtoFlags.ios =
--
78-  target-sysroot-1-darwin-ios_sim
79-
80-arch.ios_sim = x86_64
81:entrySelector.ios_sim = -alias _Konan_main _main
82-targetSysRoot.ios_sim = target-sysroot-1-darwin-ios-sim
83-libffiDir.ios_sim = libffi-3.2.1-2-darwin-ios-sim
84-llvmLtoFlags.ios_sim =
--
111-linkerDebugFlags.linux = -S
112-pluginOptimizationFlags.linux = -plugin-opt=mcpu=x86-64 -plugin-opt=O3
113-dynamicLinker.linux = /lib64/ld-linux-x86-64.so.2
114:entrySelector.linux = --defsym main=Konan_main
115-# targetSysRoot relative
116-abiSpecificLibraries.linux = ../lib64 lib64 usr/lib64
117-dependencies.linux = \
--
129-    libffi-3.2.1-2-raspberrypi
130-
131-quadruple.raspberrypi = armv6-unknown-linux-gnueabihf
132:entrySelector.raspberrypi = --defsym main=Konan_main
133-linkerOptimizationFlags.raspberrypi = --gc-sections
134-targetSysRoot.raspberrypi = target-sysroot-1-raspberrypi
135-# We could reuse host toolchain here.
--
163-    libffi-3.2.1-2-android_arm32
164-
165-quadruple.android_arm32 = arm-linux-androideabi
166:entrySelector.android_arm32 =  -Wl,--defsym -Wl,main=Konan_main
167-llvmLtoFlags.android_arm32 = -emulated-tls -relocation-model=pic
168-llvmLtoOptFlags.android_arm32 = -O3 -function-sections
169-linkerDebugFlags.android_arm32 = -Wl,-S
--
188-    libffi-3.2.1-2-android_arm64
189-
190-quadruple.android_arm64 = aarch64-linux-android
191:entrySelector.android_arm64 =  -Wl,--defsym -Wl,main=Konan_main
192-llvmLtoFlags.android_arm64 = -emulated-tls -relocation-model=pic
193-targetSysRoot.android_arm64 = target-sysroot-21-android_arm64
194-libffiDir.android_arm64 = libffi-3.2.1-2-android_arm64
--
210-linkerKonanFlags.mingw = -static-libgcc -static-libstdc++ \
211-  -Wl,-Bstatic,--whole-archive -lwinpthread -Wl,--no-whole-archive,-Bdynamic
212-linkerOptimizationFlags.mingw = -Wl,--gc-sections
213:entrySelector.mingw = -Wl,--defsym,main=Konan_main
214-dependencies.mingw = \
215-    msys2-mingw-w64-x86_64-gcc-6.3.0-clang-llvm-3.9.1-windows-x86-64 \
216-    libffi-3.2.1-mingw-w64-x86-64
```

Thank you for the support!

**Comment by olonho** [comment]
Yes, it was  problem indeed.  Shall be fixed by now.

**Comment by olonho** [comment]
Will be included into v0.3.2

**Comment by altamic** [comment]
Great, thank you.
